### ISSUE TITLE
MP campaign: "broken campaign branches" warning for players that don't have the add-on installed

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#6420]
### Game and System Information
Logging to track it, but haven't tried to reproduce it myself yet
1.16.1, reported by Bob The Mighty in https://r.wesnoth.org/p670565
See with the MP campaign Den of Thieves. 

### Describe the bug
This is an MP campaign that was designed so only the host needed the add-on. On 1.16.x players get these big warnings telling them that the campaign is broken, when it isn't. This was displayed continually to non-host players who didn't have the scenario files, but they shouldn't need them. I have got rid of these warnings by forcing a download, but it's not ideal.

**Comment by Discontinuum** [comment]
It kinda has a point that if the host disconnects, non-host players won't be able to proceed to the next scenario after finishing

**Comment by stevecotton** [comment]
> It kinda has a point that if the host disconnects, non-host players won't be able to proceed to the next scenario after finishing

Good point, but I think it still shouldn't show the message - even if the host disconnects then I believe (untested) that the start-of-scenario savefile will still be created, as it's really an end-of-previous-scenario save; so there should still be a way for players to install the add-on at that point and continue.

Also I recall a lot of push-back against adding this warning dialog, so I'm assuming there would be a consensus against showing it in any debatable edge-case.

**Comment by Pentarctagon** [comment]
Given this is a backport, is this useful to keep open?

**Comment by Wedge009** [comment]
I generally consider back-port label as relevant only to PRs. Not sure why steve has put it here - is this something that occurs in 1.18?

**Comment by stevecotton** [comment]
I haven't tested, but I assume it also affects 1.18. When I added the "backport" tag, I must have been trying to say "this also affects the stable branch".
